app/src/main/java/org/simple/clinic/searchresultsview/PatientSearchResults.kt

```diff
@@ -1,10 +1,16 @@
 package org.simple.clinic.searchresultsview
 
+import org.simple.clinic.facility.Facility
 import org.simple.clinic.patient.PatientSearchResult
 
 data class PatientSearchResults(
     val visitedCurrentFacility: List<PatientSearchResult>,
-    val notVisitedCurrentFacility: List<PatientSearchResult>
+    val notVisitedCurrentFacility: List<PatientSearchResult>,
+    val currentFacility: Facility?
 ) {
+  companion object {
+    fun emptyResults() = PatientSearchResults(emptyList(), emptyList(), null)
+  }
+
   val hasNoResults = visitedCurrentFacility.isEmpty() && notVisitedCurrentFacility.isEmpty()
```

Comment:
This is a tricky line 🤔 It works because all the properties in the data class are immutable. If that contract changes, `hasNoResults` will have its property set with the values that were used to create this class initially. A better way to protect against subtle bugs would be to make this a computed property.
```
val hasNoResults: Boolean
    get() = visitedCurrentFacility.isEmpty() && notVisitedCurrentFacility.isEmpty()
```

Reply:
Good catch. I will make this change.